test/properties.py

```diff
@@ -0,0 +1,284 @@
+from unittest import TestCase
+
+from mongoengine import connect, disconnect, DoesNotExist
+from werkzeug.security import generate_password_hash
+
+from src.controllers.properties import register_prop, all_props, update_prop, delete_prop, get_property_data
+from src.controllers.auth import register_user
+from src.models.properties import PropertyDoc
+from src.models.auth import User
+from src.utils.enums import ControllerStatus
+
+
+class CreatePropertyTests(TestCase):
+    _email = "test@example.net"
+    _password = "testpass"
+    _id = ""
+    _register_user = {
+        "username": "testuser",
+        "passwd_hash": generate_password_hash(_password),
+        "email": _email,
+        "profile": {
+            "contact_email": _email
+        }
+    }
+    _new_prop = {
+        "address": "1234 Test Street",
+        "bathrooms_amount": 1.5,
+        "bedrooms_amount": 3,
+        "contract_type": "rent",
+        "currency_code": "EUR",
+        "floors_amount": 1,
+        "garage_size": 2,
+        "price": 120000,
+        "terrain_height": 5,
+        "terrain_width": 7
+    }
+
+    def setUp(self) -> None:
+        connect("fast-home-test", host="mongomock://localhost")
+        new_user = User(**self._register_user).save()
+        self._id = str(new_user.id)
+
+    def tearDown(self) -> None:
+        disconnect()
+
+    def test_successful_create_property(self):
+        result = register_prop(self._new_prop, self._id)
+
+        self.assertEqual(result[0], ControllerStatus.SUCCESS)
+        self.assertEqual(PropertyDoc.objects.get(address=self._new_prop["address"]).address, self._new_prop["address"])
+
+
+class GetAllPropertiesTest(TestCase):
+    _email = "test@example.net"
+    _password = "testpass"
+    _id = ""
+    _register_user = {
+        "username": "testuser",
+        "passwd_hash": generate_password_hash(_password),
+        "email": _email,
+        "profile": {
+            "contact_email": _email
+        }
+    }
+    _register_prop = {
+        "address": "1234 Test Street",
+        "bathrooms_amount": 1.5,
+        "bedrooms_amount": 3,
+        "contract_type": "rent",
+        "currency_code": "EUR",
+        "floors_amount": 1,
+        "garage_size": 2,
+        "price": 120000,
+        "terrain_height": 5,
+        "terrain_width": 7
+    }
+
+    def setUp(self) -> None:
+        connect("fast-home-test", host="mongomock://localhost")
+        new_user = User(**self._register_user).save()
+        PropertyDoc(**self._register_prop).save()
+        self._id = str(new_user.id)
+
+    def tearDown(self) -> None:
+        disconnect()
+
+    def test_get_properties(self):
+        result = all_props()
+        obj = PropertyDoc.objects.get()
+
+        self.assertEqual(result[0], ControllerStatus.SUCCESS)
+        self.assertEqual(result[1].get(), obj)
```

Comment:
No, la idea es revisar que los datos de una propiedad sea igual a los datos que ingresaste
```suggestion
        self.assertEqual(result[1].first().address, self._register_prop["address"])
```

Reply:
Entonces con el ".first().address" estaríamos tomando el primer address en la bd?

Reply:
La primera propiedad de la lista y luego su dirección, pero si